src/Builder/Encoder/PipelineEncoder.php

```diff
@@ -32,7 +34,7 @@ public function encode(mixed $value): array
 
         $encoded = [];
         foreach ($value->getIterator() as $stage) {
-            $encoded[] = $this->encoder->encodeIfSupported($stage);
+            $encoded[] = $this->recursiveEncode($stage);
```

Comment:
@GromNaN: Any idea why a MixedAssignment error wasn't detected here? It was in https://github.com/mongodb/mongo-php-library/pull/1612

Reply:
I can't explain. I think this is a GitHub behavior that show the error only the first time it appears.